src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -171,12 +171,14 @@ where
             .generate::<{ IdCategory::Table }>()
             .await
             .map_err(tonic_err)? as u32;
+        source.id = id;
 
         self.catalog_manager
             .start_create_source_procedure(&source)
             .await
             .map_err(tonic_err)?;
 
+        // QUESTION(patrick): why do we need to contact compute node on create source
```

Comment:
Yes, I think we should get rid of this in the future. Source info should be part of the plan node.